mypy/treetransform.py

```diff
@@ -381,6 +382,14 @@ def visit_with_stmt(self, node: WithStmt) -> WithStmt:
         new.analyzed_types = [self.type(typ) for typ in node.analyzed_types]
         return new
 
+    def visit_match_stmt(self, o: MatchStmt) -> MatchStmt:
+        return MatchStmt(
+            subject=self.expr(o.subject),
+            patterns=o.patterns,
+            guards=self.optional_expressions(o.guards),
+            bodies=self.blocks(o.bodies),
+        )
+
```

Comment:
I wonder why the next method does not do `self.expr(node.expr)`.

Reply:
Yeah, I was wondering about that too. I guess I should probably not be lazy and write out all the boilerplate for patterns...